package/chalicelib/reportGenerator.py

```diff
@@ -42,6 +45,30 @@ def __init__(self, start_date, end_date, granularity='DAILY', metrics=None):
         self.nums_to_aliases, self.aliases_to_nums = self.build_nums_to_aliases_dicts()
         self.account_nums = list(self.nums_to_aliases.keys())
 
+        credentials = self.get_email_credentials(secret_name)
```

Comment:
I think it'd be cleaner to have this function just return the email and password as a tuple and directly assign them to the class attributes.

Reply:
I don't understand how this results in a duplicate API call.